webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -428,22 +451,166 @@ export class ContactsComponent implements OnInit, OnDestroy{
     return contact._id + contact._rev;
   }
 
-  // const setActionBarData = () => {
-  //   ctrl.setLeftActionBar({
-  //     hasResults: ctrl.hasContacts,
-  //     userFacilityId: usersHomePlace && usersHomePlace._id,
-  //     childPlaces: allowedChildPlaces,
-  //     exportFn: function() {
-  //       Export('contacts', ctrl.filters, { humanReadable: true });
-  //     },
-  //   });
-  // };
-
-  // const updateAllowedChildPlaces = () => {
-  //   XmlForms.listen('ContactsListCtrl',{ contactForms: true }, (err, forms) => {
-  //     const allowCreateLink = contactType => forms && forms.find(form => form._id === contactType.create_form);
-  //     allowedChildPlaces = childPlaces.filter(allowCreateLink);
-  //     setActionBarData();
-  //   });
-  // };
+  private setLeftActionBar() {
+    this.globalActions.setLeftActionBar({
+      hasResults: this.hasContacts,
+      userFacilityId: this.usersHomePlace && this.usersHomePlace._id,
+      childPlaces: this.allowedChildPlaces?.sort((a, b) => a.id.localeCompare(b.id)),
+      exportFn: () => {
+        this.exportService.export('contacts', this.filters, { humanReadable: true });
+      }
+    });
+  }
+
+  private subscribeToSelectedContact() {
+    const subscription = this.store
+      .select(Selectors.getSelectedContact)
+      .subscribe((selected) => {
+        this.selectedContact = selected;
+        this.setChildTypesBySelectedContact();
+
+        this.globalActions.setRightActionBar({
+          relevantForms: [], // This disables the "New Action" button in action bar until forms load
+          sendTo: this.selectedContact?.type?.person ? this.selectedContact?.doc : '',
+          canDelete: !!this.selectedContact?.children?.every(group => !group.contacts?.length),
+          canEdit: this.sessionService.isAdmin() || this.userSettings?.facility_id === this.selectedContact?.doc?._id,
```

Comment:
`this.userSettings?.facility_id === this.selectedContact?.doc?._id`
this means that an offline user can ONLY edit their home place?

This is exactly the opposite as the original code: 
```
 return userSettings.facility_id && userSettings.facility_id !== selected.doc._id;
```
where an offline user can edit _any_ contact EXCEPT their home place. 

I'm betting there are no unit tests around this. I think we should add some.

Reply:
Oh yeah Njuguna reported the issue, I'll fix it

Reply:
I think my eyes betrayed me, changed the operator, fixed and added unit test